subsys/testsuite/ztest/Kconfig

```diff
@@ -91,6 +91,25 @@ config ZTEST_RULE_1CPU
 
 endmenu
 
+config ZTEST_DO_THE_SHUFFLE
+	bool "Enables the shuffling of tests and suites"
+	default n
+	select TEST_RANDOM_GENERATOR if !ENTROPY_HAS_DRIVER
+	help
+	  This rule will shuffle the order of tests and test suites.
+
+config ZTEST_DO_THE_SHUFFLE_SUITE_REPEAT_COUNT
+	int "Number of iterations the test suite will run"
+	default 3
+	help
+	  This rule will execute a test suite N number of times.
+
+config ZTEST_DO_THE_SHUFFLE_TEST_REPEAT_COUNT
+	int "Number of iterations the test will run"
+	default 3
+	help
+	  This rule will execute a test N number of times.
```

Comment:
Same here, more details would be great. Tests will be shuffled within the same suite.